seam/client.py

```diff
@@ -28,8 +28,11 @@ def __init__(self, base_url: str, auth_headers: Dict[str, str], **kwargs):
         self.headers.update(headers)
 
     def request(self, method, url, *args, **kwargs):
-        url = urljoin(self.base_url, url)
-        response = super().request(method, url, *args, **kwargs)
+        full_url = urljoin(self.base_url, url)
+        headers = {**self.headers, **(kwargs.get("headers", {}) or {})}
+        kwargs["headers"] = headers
```

Comment:
What is this change for? Do we still need it if we can just use `Seam().client` instead?

Reply:
So the test mocks `request` method and checks what headers were passed through params. This change explicitly passes the headers so that we can access them from `mock_request.call_args`. Without this change, call args won't have headers since they are not passed but taken from `self.headers`. I tried intercepting the request similarly to what JS SDK does with `nock` but I couldn't make it work; the request kept hitting the real endpoint. Lmk if you have some ideas.

Reply:
I see. Seems like this should be safe according to https://stackoverflow.com/questions/45883655/is-it-always-safe-to-modify-the-kwargs-dictionary

Reply:
I was going to suggest adding a comment like
```
        full_url = urljoin(self.base_url, url)
        # UPSTREAM: Even though niquest will use self.headers, the tests only mock the request method and ...
        headers = {**self.headers, **(kwargs.get("headers", {}) or {})}
        kwargs["headers"] = headers
```

But what if we just make the tests understand this limitation instead of runtime, e.g., can we do something like this?

```
        assert "headers" in mock_request.call_args.kwargs
        passed_headers = {**client.headers, **mock_request.call_args.kwargs["headers"]}
```

Reply:
Not sure I understand. Are you proposing instead of passing headers explicitly, use:
```py
        assert "headers" in mock_request.call_args.kwargs
        passed_headers = {**client.headers, **mock_request.call_args.kwargs["headers"]}
```
If so then I think this won't work as `mock_request.call_args.kwargs["headers"]` will be `None` if `headers` are not assigned explicitly to `kwargs` and passed to `request`

Reply:
Yea, I'm saying the test can replicate the known behavior of the lib, where it combines the passed headers with the ones in `self.headers`. Neither solution is perfect but IMO the runtime code should not compromise for the test. If it's none, we can do

```
        passed_headers = mock_request.call_args.kwargs.get("headers", {})
        request_headers = {**client.headers, **passed_headers}
# assert stuff on request_headers
```

Reply:
bdddf939419a19ca8fb3014586c12c42dd2534f2